### PR TITLE
[centec]update centec e582 platform drivers and libsai

### PR DESCRIPTION
**Pull request by yangbashuang**
1. update platform drivers to new kernel
2. update libsai to 1.3.3
3. update platform device data and config file
4. add support e582 48x2q4z board
**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: platform/centec/sonic-platform-modules-e582/48x2q4z/modules/centec_at24c64.c [hunk 1]
```diff
@@ -1,657 +1,602 @@
...
+    ctc_at24c64_exit();
+    i2c_del_driver(&at24_ctc_driver);
+}
+module_exit(at24_ctc_exit);
+
+MODULE_DESCRIPTION("Driver for most I2C EEPROMs");
+MODULE_AUTHOR("David Brownell and Wolfram Sang");
+MODULE_LICENSE("GPL");
+/* XXX */
+
```
Comment (lguohan) [inline platform/centec/sonic-platform-modules-e582/48x2q4z/modules/centec_at24c64.c]: why not use standard at24 kernel driver?

Comment (yangbashuang) [reply platform/centec/sonic-platform-modules-e582/48x2q4z/modules/centec_at24c64.c]: hi lguohan:
thanks for review
i created a eeprom sys file in private driver for user to read and get eeprom data
/sys/class/i2c-adapter/i2c-0/0-0057/eeprom this file is created in this private driver

i tried to use ac24 kernel standard driver, could not found eeprom sys file to read eeprom data, did i miss something, please help me how to do it with standard at24 driver?

i see 
 sonic-buildimage/platform/broadcom/sonic-platform-modules-cel/dx010/modules/mc24lc64t.c
use the same way as i do

**Comment by yangbashuang** [comment]
@lguohan
@yxieca 
please merge this PR, thanks

File: platform/centec/sonic-platform-modules-e582/48x2q4z/cfg/minigraph.xml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<DeviceMiniGraph xmlns:i="http://www.w3.org/2001/XMLSchema-instance" xmlns="Microsoft.Search.Autopilot.Evolution">
```
Comment (nikos-github) [inline platform/centec/sonic-platform-modules-e582/48x2q4z/cfg/minigraph.xml]: Why do you need the minigraph.xml file?

**Comment by yangbashuang** [comment]
@nikos-github minigraph.xml removed, please review, thanks

File: device/centec/x86_64-centec_e582_48x2q4z-r0/plugins/psuutil.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+
+#############################################################################
+# Mellanox
```
Comment (nikos-github) [inline device/centec/x86_64-centec_e582_48x2q4z-r0/plugins/psuutil.py]: Mellanox?

**Comment by yangbashuang** [comment]
@nikos-github update psuutil.py comments , please review, thanks

File: device/centec/x86_64-centec_e582_48x2q4z-r0/E582-48x2q4z/qos.json [hunk 4]
```diff
@@ -0,0 +1,133 @@
+{
```
Comment (lguohan) [inline device/centec/x86_64-centec_e582_48x2q4z-r0/E582-48x2q4z/qos.json]: we have unified qos.json.j2 now, can you check other platform?

**Comment by lguohan** [comment]
can you use minigraph.xml and use default_sku as other platforms?

File: device/centec/x86_64-centec_e582_48x2q4z-r0/E582-48x2q4z/qos.json (hunk 4 above)
Comment (lguohan) [reply device/centec/x86_64-centec_e582_48x2q4z-r0/E582-48x2q4z/qos.json]: https://github.com/Azure/sonic-buildimage/blob/master/device/arista/x86_64-arista_7050_qx32/Arista-7050-QX32/qos.json.j2

File: platform/centec/python-saithrift.mk [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # python-saithrift package
 
-PYTHON_SAITHRIFT_CENTEC = python-saithrift_1.2.1_amd64.deb
+PYTHON_SAITHRIFT_CENTEC = python-saithrift_1.3.0_amd64.deb
```
Comment (lguohan) [inline platform/centec/python-saithrift.mk]: 1.3.5 now.

File: platform/centec/libsaithrift-dev.mk [hunk 6]
```diff
@@ -1,6 +1,6 @@
 # libsaithrift-dev package
 
-LIBSAITHRIFT_DEV_CENTEC = libsaithrift-dev_1.2.1_amd64.deb
+LIBSAITHRIFT_DEV_CENTEC = libsaithrift-dev_1.3.0_amd64.deb
```
Comment (lguohan) [inline platform/centec/libsaithrift-dev.mk]: 1.3.5

File: platform/centec/sonic-platform-modules-e582/debian/platform-modules-e582-48x2q4z.install [hunk 7]
```diff
@@ -1,2 +1,7 @@
 48x2q4z/cfg/48x2q4z-modules.conf etc/modules-load.d
+48x2q4z/cfg/minigraph.xml etc/sonic
```
Comment (nikos-github) [inline platform/centec/sonic-platform-modules-e582/debian/platform-modules-e582-48x2q4z.install]: Since minigraph.xml has been removed, is this needed?

**Comment by yangbashuang** [comment]
@lguohan 
@nikos-github 
thanks for reviewing, changes requested are done, please help check again

**Pull request merged**